app/models/identity.rb

```diff
@@ -39,6 +39,10 @@ def self.build_from_omniauth(provider)
   end
 
   def email
-    auth_data_dump.info.email
+    if auth_data_dump&.info&.email
+      auth_data_dump.info.email
+    else
+      "No email found. Please relink your #{provider} account to avoid errors."
+    end
```

Comment:
I think you can refactor this a bit, since using `&.` returns `nil` early on. Maybe turn this into one line, and use a guard statement with an `unless` instead?

Reply:
Thank you @Zhao-Andy for quick reply with good suggestions.
I've implemented a guard clause.